src/rest_vol.c

```diff
@@ -2307,7 +2360,209 @@ RV_find_object_by_path(RV_object_t *parent_obj, const char *obj_path, H5I_type_t
+                            url_encoded_path_name, is_relative_path ? "&parent_id=" : "",
+                            is_relative_path ? parent_obj->URI : "")) < 0)
+        FUNC_GOTO_ERROR(H5E_LINK, H5E_SYSERRSTR, FAIL, "snprintf error");
+
+    if (url_len >= URL_MAX_LENGTH)
+        FUNC_GOTO_ERROR(H5E_LINK, H5E_SYSERRSTR, FAIL, "link GET request URL size exceeded maximum URL size");
+
+#ifdef RV_CONNECTOR_DEBUG
+    printf("-> Retrieving link type for link to target object at URL %s\n\n", request_url);
+#endif
+
+    /* Setup cURL for making GET requests */
+
+    /* Setup the host header */
+    host_header_len = strlen(parent_obj->domain->u.file.filepath_name) + strlen(host_string) + 1;
+    if (NULL == (host_header = (char *)RV_malloc(host_header_len)))
+        FUNC_GOTO_ERROR(H5E_LINK, H5E_CANTALLOC, FAIL, "can't allocate space for request Host header");
+
+    strcpy(host_header, host_string);
+
+    curl_headers =
+        curl_slist_append(curl_headers, strncat(host_header, parent_obj->domain->u.file.filepath_name,
+                                                host_header_len - strlen(host_string) - 1));
+
+    /* Disable use of Expect: 100 Continue HTTP response */
+    curl_headers = curl_slist_append(curl_headers, "Expect:");
+
+    if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_HTTPHEADER, curl_headers))
+        FUNC_GOTO_ERROR(H5E_LINK, H5E_CANTSET, FAIL, "can't set cURL HTTP headers: %s", curl_err_buf);
+    if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_HTTPGET, 1))
+        FUNC_GOTO_ERROR(H5E_LINK, H5E_CANTSET, FAIL, "can't set up cURL to make HTTP GET request: %s",
+                        curl_err_buf);
+    if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_URL, request_url))
+        FUNC_GOTO_ERROR(H5E_LINK, H5E_CANTSET, FAIL, "can't set cURL request URL: %s", curl_err_buf);
+
+#ifdef RV_CONNECTOR_DEBUG
+    printf("   /**********************************\\\n");
+    printf("-> | Making GET request to the server |\n");
+    printf("   \\**********************************/\n\n");
+#endif
+
+    CURL_PERFORM(curl, H5E_LINK, H5E_PATH, FALSE);
+
+    if (CURLE_OK != curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_response))
+        FUNC_GOTO_ERROR(H5E_LINK, H5E_CANTGET, FAIL, "can't get HTTP response code");
+
+    ret_value = HTTP_SUCCESS(http_response);
+
+#ifdef RV_CONNECTOR_DEBUG
+    printf("-> Object %s\n\n", ret_value ? "found" : "not found");
+#endif
+
+    if (ret_value > 0) {
+
+        if (obj_found_callback && RV_parse_response(response_buffer.buffer, callback_data_in,
+                                                    callback_data_out, obj_found_callback) < 0) {
+            FUNC_GOTO_ERROR(H5E_OBJECT, H5E_CALLBACK, FAIL, "can't perform callback operation");
+        }
+
+    } /* end if */
+
+    char uri[URI_MAX_LENGTH];
+
+    if (0 > RV_parse_response(response_buffer.buffer, NULL, uri, RV_copy_object_URI_callback))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_CANTGET, FAIL, "failed to get URI");
```

Comment:
It looks like this URI object doesn't get used anywhere after being parsed out of the reponse